### PR TITLE
#58 Implement presentation unsubscription

### LINKED ISSUE #58: Leave presentation
Create laravel controllers for leaving a presentation

**Estimation: 1h**

### PR DESCRIPTION
**Pull request by kayoumido**

### CODE REVIEW

File: src/app/Http/Controllers/API/PresentationUserController.php [hunk 1]
```diff
@@ -39,7 +39,24 @@ public function subscribe(Presentation $presentation, User $user)
...
     public function unsubscribe(Presentation $presentation, User $user)
     {
-        //
+        // check that the user isn't already unsubscribed to the presentation
+        if (!$presentation->users()->find($user)) {
+            return response()->json([
+                'success' => false,
+                'message' => "Already unsubscribed",
+            ], 409);
+        }
```
Comment (fabio10000) [inline src/app/Http/Controllers/API/PresentationUserController.php]: No need to to check if user is subscribed to the presentation as if it is not the detach will simply not do anything

File: src/app/Http/Controllers/API/PresentationUserController.php [hunk 2]
```diff
@@ -39,7 +39,24 @@ public function subscribe(Presentation $presentation, User $user)
...
+        // all gucci, the user can unsubscribe
+
+        // unsubscribe the user to the presentation
+        $presentation->users()->detach($user);
+
+        // return success
+        return response()->json([
+            'success' => true,
+            'message' => "Successfully unsubscribed",
+        ], 200);
```
Comment (fabio10000) [inline src/app/Http/Controllers/API/PresentationUserController.php]: An HTTP 200 is automaticly sent if there is no errors produced along the function, so no need of that code

**Review by fabio10000 — CHANGES_REQUESTED** [review]
You need to check if the user that is unsubscribing is the connected user or an admin of the presentation to avoid request forgery, you could create a middleware for that logic so it could be reused on other controllers

File: src/app/Http/Controllers/API/PresentationUserController.php (hunk 1 above)
Comment (kayoumido) [reply src/app/Http/Controllers/API/PresentationUserController.php]: it was mainly to tell the user he's an idiot and he already unsubscribed

File: src/app/Http/Controllers/API/PresentationUserController.php (hunk 2 above)
Comment (kayoumido) [reply src/app/Http/Controllers/API/PresentationUserController.php]: okay

File: src/app/Http/Controllers/API/PresentationUserController.php (hunk 1 above)
Comment (fabio10000) [reply src/app/Http/Controllers/API/PresentationUserController.php]: Ok so you should use the Response class to send the HTTP status so it's more readable you can use it like `Response::HTTP_CONFLICT` or something like that

**Review by fabio10000 — CHANGES_REQUESTED** [review]
Change concerning the response status codes see previous comment

File: src/app/Http/Controllers/API/PresentationUserController.php [hunk 3]
```diff
@@ -41,7 +42,24 @@ public function subscribe(Presentation $presentation, User $user)
...
+        // all gucci, the user can unsubscribe
+
+        // unsubscribe the user to the presentation
+        $presentation->users()->detach($user);
+
+        // return success
+        return response()->json([
+            'success' => true,
+            'message' => "Successfully unsubscribed",
+        ], Response::HTTP_OK);
```
Comment (fabio10000) [inline src/app/Http/Controllers/API/PresentationUserController.php:62]: No need but ok

**Review by fabio10000 — APPROVED** [review]

**Review by alfavre — APPROVED** [review]
tested and could unsubscribe from presentation, however it only worked with presentator as I could not subscribe any other user, I approve so that it can be merged with subscribe #56 and then I will test it again

**Comment by kayoumido** [comment]
Will be validated in #118

**Pull request closed without merging**